### PR TITLE
Link with cblas when LLAMA_OPENBLAS is enabled.

### LINKED ISSUE #634: Compiling with LLAMA_OPENBLAS=1 fails
The problem is missing `cblas` symbols:

```plaintext
g++ -I. -I./examples -O3 -DNDEBUG -std=c++11 -fPIC -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -pthread examples/main/main.cpp ggml.o llama.o common.o -o main -lopenblas
/usr/bin/ld: ggml.o: in function `ggml_compute_forward_mul_mat_f16_f32':
ggml.c:(.text+0x454b): undefined reference to `cblas_sgemm'
/usr/bin/ld: ggml.o: in function `ggml_compute_forward':
ggml.c:(.text+0xcc1b): undefined reference to `cblas_sgemm'
/usr/bin/ld: ggml.c:(.text+0xd42d): undefined reference to `cblas_sgemm'
collect2: error: ld returned 1 exit status
make: *** [Makefile:241: main] Error 1
```

Installing `cblas` and adding `-lcblas` to the link flags fixes the issue.

This is on Arch Linux. I'll also submit the trivial PR, but I don't know if there are different versions/systems where this would cause issues.

### PR DESCRIPTION
**Pull request by KerfuffleV2**
Please see #634 for the details about what this is intending to fix.
TL;DR: At least on Archlinux, linking when BLAS mode is on fails due to missing `cblas` symbols.

Closes #634

### CODE REVIEW

**Comment by slaren** [comment]
This breaks the build on Ubuntu.

**Comment by slaren** [comment]
Yes, it compiles fine as it is now.

**Comment by cyyynthia** [comment]
On Arch Linux, I had to use the https://aur.archlinux.org/packages/openblas-lapack package instead of simply openblas, and it solved my problems with cblas.

**Comment by KerfuffleV2** [comment]
I guess I'll just close this since it seems like Archlinux is weird weird. Ideally there would be some way to detect what combination of flags is necessary but llama.cpp doesn't really seem to have a `./configure` type build system that can do fancy stuff like that at the moment.

1. https://bugs.archlinux.org/task/59046
2. https://bugs.archlinux.org/task/66092

One is from 2018 and the other is from 2020 so I wouldn't hold my breath hoping this gets fixed on the Arch distro side.

**Comment by loretoparisi** [comment]
> On Arch Linux, I had to use the https://aur.archlinux.org/packages/openblas-lapack package instead of simply openblas, and it solved my problems with cblas.

I tried this way but I get the same error `ggml.c:(.text+0x13b20): undefined reference to `cblas_sgemm`

```bash
wget https://github.com/xianyi/OpenBLAS/releases/download/v0.3.23/OpenBLAS-0.3.23.tar.gz
tar -xvzf OpenBLAS-0.3.23.tar.gz 
cd OpenBLAS-0.3.23
make
make PREFIX=/usr/local/include/openblas install
cd ../
cd llama.cpp/
make clean
make LLAMA_OPENBLAS=1
I llama.cpp build info: 
I UNAME_S:  Linux
I UNAME_P:  unknown
I UNAME_M:  aarch64
make: pkg-config: No such file or directory
I CFLAGS:   -I.            -O3 -std=c11   -fPIC -DNDEBUG -Wall -Wextra -Wpedantic -Wcast-qual -Wdouble-promotion -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Wno-unused-function -pthread -mcpu=native -DGGML_USE_K_QUANTS -DGGML_USE_OPENBLAS 
I CXXFLAGS: -I. -I./common -O3 -std=c++11 -fPIC -DNDEBUG -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-multichar -Wno-format-truncation -pthread -mcpu=native -DGGML_USE_K_QUANTS
make: pkg-config: No such file or directory
I LDFLAGS:  
I CC:       cc (Debian 10.2.1-6) 10.2.1 20210110
I CXX:      g++ (Debian 10.2.1-6) 10.2.1 20210110

make: pkg-config: No such file or directory
g++ -I. -I./common -O3 -std=c++11 -fPIC -DNDEBUG -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-multichar -Wno-format-truncation -pthread -mcpu=native -DGGML_USE_K_QUANTS examples/main/main.cpp ggml.o llama.o common.o console.o grammar-parser.o k_quants.o ggml-alloc.o -o main 
/usr/bin/ld: ggml.o: in function `ggml_compute_forward_mul_mat':
ggml.c:(.text+0x13b20): undefined reference to `cblas_sgemm'
collect2: error: ld returned 1 exit status
make: *** [Makefile:427: main] Error 1
```

**Comment by Cebtenzzre** [comment]
> I tried this way but I get the same error

It looks like you need to install pkg-config.
